windows/devices/uwb/UwbConnector.cxx

```diff
@@ -943,36 +1137,29 @@ UwbConnector::DeregisterEventCallback(std::weak_ptr<::uwb::RegisteredCallbackTok
     }
     std::lock_guard eventCallbacksLockExclusive{ m_eventCallbacksGate };
 
+    auto callbacksPresentPrior = CallbacksPresent();
+
     auto sessionCallback = dynamic_pointer_cast<::uwb::RegisteredSessionCallbackToken>(tokenShared);
     if (sessionCallback) {
         // treat it as a session callback
         auto sessionId = sessionCallback->SessionId;
 
-        auto node = m_sessionEventCallbacks.extract(sessionId);
-        if (node.empty()) {
-            return; // sessionId has no associated tokens
-        }
+        DeregisterSessionEventCallback<::uwb::OnSessionEndedToken>(tokenShared, m_onSessionEndedCallbacks) or
+            DeregisterSessionEventCallback<::uwb::OnRangingStartedToken>(tokenShared, m_onRangingStartedCallbacks) or
+            DeregisterSessionEventCallback<::uwb::OnRangingStoppedToken>(tokenShared, m_onRangingStoppedCallbacks) or
+            DeregisterSessionEventCallback<::uwb::OnPeerPropertiesChangedToken>(tokenShared, m_onPeerPropertiesChangedCallbacks) or
+            DeregisterSessionEventCallback<::uwb::OnSessionMembershipChangedToken>(tokenShared, m_onSessionMembershipChangedCallbacks);

```

Comment:
It'd be nice if we could avoid this pattern. The type information is built in to the instance (`tokenShared`), so we could make use of the polymorphic nature of the token and have it de-register itself. For example, you could add a pure `virtual` method to `RegisteredSessionCallbackToken` like:

```
struct RegisteredSessionCallbackToken {
    virtual void Deregister() = 0;
```

Then have sub-classes implement that. So, something like this:
```
struct OnSessionEndedToken : public RegisteredSessionCallbackToken
{
    OnSessionEndedToken(uint32_t sessionId, std::weak_ptr<::uwb::UwbRegisteredSessionEventCallbackTypes::OnSessionEnded> callback, std::unordered_map<uint32_t, std::vector<std::shared_ptr<::uwb::OnSessionEndedToken>>> &tokenMap) :
        RegisteredSessionCallbackToken(sessionId),
        Callback(std::move(callback)){};
    std::weak_ptr<::uwb::UwbRegisteredSessionEventCallbackTypes::OnSessionEnded> Callback;
    std::unordered_map<uint32_t, std::vector<std::shared_ptr<::uwb::OnSessionEndedToken>>>& TokenMap;
    
    void Deregister() override { DeregisterSessionEventCallback(this, TokenMap);
};
```

This would require updating `DeregisterSessionEventCallback` to take a `::uwb::RegisteredCallbackToken*` instead of the `shared_ptr<::uwb::RegisteredCallbackToken>`, but that's no issue at all.

I thought it would also be nice to have it encode the ability to execute the callback as well using the same pattern. Anyway, perhaps something to consider for a follow-up PR.

Reply:
resolving this to be addressed in the next pr